modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/test/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SearchIndexResourceImplTest.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.search.experiences.rest.internal.resource.v1_0;
+
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.search.index.IndexInformation;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+import com.liferay.search.experiences.rest.dto.v1_0.SearchIndex;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Wade Cao
+ */
+public class SearchIndexResourceImplTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() throws Exception {
+		_searchIndexResourceImpl = new SearchIndexResourceImpl();
+
+		_searchIndexResourceImpl.setContextCompany(_contextCompany);
+
+		ReflectionTestUtil.setFieldValue(
+			_searchIndexResourceImpl, "_indexInformation", _indexInformation);
+	}
+
+	@Test
+	public void testGetSearchIndexes() {
+		_setUpContextCompany(12345);
+		_setUpIndexInformation(
+			"prod-12345",
+			new String[] {
+				"liferay-12345-search-tuning-xyz", "prod-0", "prod-12345",
+				"prod-12345-search-tuning-rankings",
+				"prod-12345-search-tuning-synonyms",
+				"prod-54321-search-tuning-xyz"
+			});
+
+		List<SearchIndex> searchIndexes =
+			_searchIndexResourceImpl.getSearchIndexes();
+
+		Assert.assertEquals(searchIndexes.toString(), 2, searchIndexes.size());
+
+		SearchIndex searchIndex = searchIndexes.get(0);
+
+		Assert.assertEquals(
+			"prod-12345-search-tuning-rankings", searchIndex.getName());
+		Assert.assertEquals(
+			"12345-search-tuning-rankings", searchIndex.getSuffix());
```

Comment:
@wcao20170619 shouldnt the suffix just be "search-tuning-rankings"?

Reply:
"companyId-" + search-tuning-rankings

Reply:
search-tuning-rankings belong to a company. without a companyId we don't know what company this indexname belongs to. That is the question I asked you before and @lipusz answered. and let me talk to you - https://issues.liferay.com/browse/LPS-163750

Reply:
right, i dont fully know what the suffix is needed for... from what i could tell, it was for import/export. so if we created a blueprint to search rankings, it could be imported to multiple companies... therefore we only need the suffix, without companyId.... but maybe im misunderstanding

Reply:
ya i will wait for tibors response, but its probably safer to include the companyId like you are doing here. i still dont fully understand where/how we are using the suffix, but i think its easier to remove the companyId from the suffix later on if we dont need it that to put it back if we do need it

Reply:
Thx! It will be used in search-experience-service and do something like searchRequestBuilder.addIndex(fullIndexName); I am currently writing an integration test to make sure it works.